test/language/block-scope/syntax/for-in/disallow-initialization-assignment.js

```diff
@@ -0,0 +1,11 @@
+// Copyright (C) Copyright 2011 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 13.1
+description: >
+    for declaration:
+    disallow initialization assignment
+negative: SyntaxError
+---*/
+for (let x = 3 in {}) { } }
```

Comment:
The trailing `}` means that this file will always yield a syntax error, even in runtimes that ought to fail (i.e. those that allow initialization assignments as the left hand side of a for..in statement)